tools/bug-report/pkg/bugreport/bugreport.go

```diff
@@ -0,0 +1,262 @@
+		if err != nil {
+			log.Error(err.Error())
+			continue
+		}
+
+		switch {
+		case container == "istio-proxy":
+			getFromCluster(content.GetCoredumps, &content.Params{DryRun: config.DryRun, Namespace: namespace, Pod: pod, Container: container},
+				archive.ProxyCoredumpPath(tempDir, namespace, pod), &mandatoryWg)
+			getProxyLogs(config, resources, p, namespace, pod, container, &optionalWg)
+
+		case strings.HasPrefix(pod, "istiod-") && container == "discovery":
+			getFromCluster(content.GetIstiodInfo, &content.Params{DryRun: config.DryRun, Namespace: namespace, Pod: pod, Container: container},
+				archive.IstiodPath(tempDir, namespace, pod), &mandatoryWg)
+			getIstiodLogs(config, resources, namespace, pod, &mandatoryWg)
+
+		}
+	}
+
+	// Not all items are subject to timeout. Proceed only if the non-cancellable items have completed.
+	mandatoryWg.Wait()
+
+	// If log fetches have completed, cancel the timeout.
+	go func() {
+		optionalWg.Wait()
+		cmdTimer.Stop()
+	}()
+
+	// Wait for log fetches, up to the timeout.
+	<-cmdTimer.C
+}
+
+// getFromCluster runs a cluster info fetching function f against the cluster and writes the results to fileName.
+// Runs if a goroutine, with errors reported through gErrors.
+func getFromCluster(f func(params *content.Params) (map[string]string, error), params *content.Params, dir string, wg *sync.WaitGroup) {
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		out, err := f(params)
+		appendGlobalErr(err)
+		if err == nil {
+			writeFiles(dir, out)
+		}
+	}()
+}
+
+// getProxyLogs fetches proxy logs for the given namespace/pod/container and stores the output in global structs.
+// Runs if a goroutine, with errors reported through gErrors.
+// TODO(stewartbutler): output the logs to a more robust/complete structure.
+func getProxyLogs(config *config.BugReportConfig, resources *cluster2.Resources, path, namespace, pod, container string, wg *sync.WaitGroup) {
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		clog, cstat, imp, err := getLog(resources, config, namespace, pod, container)
+		appendGlobalErr(err)
+		lock.Lock()
+		if err == nil {
+			logs[path], stats[path], importance[path] = clog, cstat, imp
+		}
+		lock.Unlock()
+	}()
+}
+
+// getIstiodLogs fetches Istiod logs for the given namespace/pod and writes the output.
+// Runs if a goroutine, with errors reported through gErrors.
+func getIstiodLogs(config *config.BugReportConfig, resources *cluster2.Resources, namespace, pod string, wg *sync.WaitGroup) {
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		clog, _, _, err := getLog(resources, config, namespace, pod, "discovery")
+		appendGlobalErr(err)
+		writeFile(archive.IstiodPath(tempDir, namespace, pod+".log"), clog)
+	}()
+}
+
+// getLog fetches the logs for the given namespace/pod/container and returns the log text and stats for it.
+func getLog(resources *cluster2.Resources, config *config.BugReportConfig, namespace, pod, container string) (string, *processlog.Stats, int, error) {
```

Comment:
I have had problems where one pod has a truely gigantic log and the other pods have reasonable logs.  It might be a good idea to pass `--since`, or `--limit-bytes`, or `--tail`

Reply:
afaik there's no easy way to limit how much of the log you fetch this way. we do have limits in the top level command (both size and time) and these work to discard logs or kill off any goroutines that are still running.

Reply:
you can limit the number of lines with `--tail` on kubectl

Reply:
+1, I think --since would be useful and kubectl log does that as well